manifests/manifests.go

```diff
@@ -135,6 +137,66 @@ func Read(manifestPaths []string, dependOnSelf bool) ([]string, Dependencies, []
 	return components, Dependencies{dependencies}, nil
 }
 
+// ReadRepoManifest reads a full repository manifest as produced by monobuild print --full
+func ReadRepoManifest(manifest string, dependOnSelf bool) ([]string, Dependencies, []error) {
+	lines := strings.Split(manifest, "\n")
+	dependencies := make(map[string][]Dependency, len(lines))
+	components := make([]string, 0, len(lines))
+	errors := []error{}
+
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if len(line) < 1 || line[0] == '#' { // skip blank lines and comments
+			continue
+		}
+
+		parts := strings.Split(line, ":")
+		if len(parts) != 2 {
+			errors = append(errors, fmt.Errorf("bad line format: '%s' expected 'componnennt: dependency, dependency, ...'", line))
+			continue
+		}
+
+		component := strings.TrimSpace(parts[0])
+		components = append(components, component)
+
+		dependencies[component] = []Dependency{}
+
+		for _, d := range strings.Split(parts[1], ",") {
+			if len(strings.TrimSpace(d)) < 1 {
+				continue
+			}
+
+			dep, err := readDependency(d)
+			if err != nil {
+				errors = append(errors, fmt.Errorf("malformed dependency: %s", d))
+				continue
+			}
+
+			dependencies[component] = append(dependencies[component], dep)
+		}
+
+		if dependOnSelf {
+			dependencies[component] = append([]Dependency{Dependency{component, Weak}}, dependencies[component]...)
+		}
+	}
+
+	// validate dependencies
+	for manifest, deps := range dependencies {
+		for _, dep := range deps {
+			if !validDependency(components, dep) {
+				errors = append(errors, fmt.Errorf("unknown dependency '%s' of '%s'", dep.Name, manifest))
+			}
+		}
+	}
+
+	if len(errors) > 0 {
+		return nil, Dependencies{dependencies}, errors
```

Comment:
Why not `nil, nil, errors` ? I suppose nobody would like to use the invalid `Dependencies{dependencies}` anyway ?

Reply:
Good spot, it was supposed to say `Dependencies{}` - an empty value for the `Dependencies` type. Go won't let me use nil for a value of that type.